soda/core/soda/execution/derived_formula.py

```diff
@@ -5,3 +5,12 @@ class DerivedFormula:
     def __init__(self, function: Callable, metric_dependencies: Dict[str, "Metric"]):
         self.function: Callable = function
         self.metric_dependencies: Dict[str, "Metric"] = metric_dependencies
+
+    def collect_failed_rows_sample_refs(self):
+        if isinstance(self.metric_dependencies, dict):
```

Comment:
When would this be _not_ an instance of `dict` ? I mean - why check for object type instead of value?

Reply:
No particular reason.  I start to use isinstance more since just checking for value has bitten me couple of times.

Reply:
ok, thanks!